ladybug_geometry/_mesh.py

```diff
@@ -0,0 +1,323 @@
+
+        When colors are assigned to the mesh, this property is linked to the colors
+        and cannot be set. However, when no colors are assigned, it can be set
+        because it affects how the mesh is translated to external interfaces.
+        In such cases, this property notes whether the mesh will be translated
+        in a format that can accept face-by-face colors.
+        Default is False when no colors are assigned.
+        """
+        return self._is_color_by_face
+
+    @is_color_by_face.setter
+    def is_color_by_face(self, by_face):
+        assert isinstance(by_face, bool), \
+            'is_color_by_face must be a boolean. Got {}'.format(type(by_face))
+        if self._colors is not None and self._is_color_by_face is not by_face:
+            raise AttributeError('is_color_by_face cannot be set when colors are'
+                                 ' already assinged to the mesh.')
+        self._is_color_by_face = by_face
+
+    @property
+    def area(self):
+        """The area of the entire mesh."""
+        if self._area is None:
+            self._area = sum(self.face_areas)
+        return self._area
+
+    @property
+    def face_centroids(self):
+        """A tuple of face centroids that parallels the Faces property."""
+        if self._face_centroids is None:
+            _f_cent = []
+            for face in self.faces:
+                if len(face) == 3:
+                    _f_cent.append(self._tri_face_centroid(face))
+                else:
+                    _f_cent.append(self._quad_face_centroid(face))
+            self._face_centroids = tuple(_f_cent)
+        return self._face_centroids
+
+    def move(self, moving_vec):
+        """Get a mesh that has been moved along a vector.
+
+        Args:
+            moving_vec: A Vector with the direction and distance to move the mesh.
+        """
+        _verts = tuple(pt.move(moving_vec) for pt in self.vertices)
+        return self._mesh_transform(_verts)
+
+    def reflect(self, normal, origin):
+        """Get a mesh reflected across a plane with the input normal vector and origin.
+
+        Args:
+            normal: A Vector representing the normal vector for the plane across
+                which the mesh will be reflected. THIS VECTOR MUST BE NORMALIZED.
+            origin: A Point representing the origin from which to reflect.
+        """
+        _verts = tuple(pt.reflect(normal, origin) for pt in self.vertices)
+        return self._mesh_transform(_verts)
+
+    def scale(self, factor, origin):
+        """Scale a mesh by a factor from an origin point.
+
+        Args:
+            factor: A number representing how much the mesh should be scaled.
+            origin: A Point representing the origin from which to scale.
+        """
+        _verts = tuple(pt.scale(factor, origin) for pt in self.vertices)
+        return self._mesh_scale(_verts, factor)
+
+    def scale_world_origin(self, factor):
```

Comment:
I would like to keep this in here because `scale_world_origin` will be much faster when you want to scale an large model and you do not care about the origin.  This is because it completely ignores the addition and subtraction of the origin point for each vertex and essentially just multiplies the vertex xyz values by the factor.  I am thinking that it will be particularly helpful for enabling people to build Honeybee energy models in different unit systems.  Right now in legacy HB, if you build a larger multizone model in a system like feet, you can end up waiting a few minutes for it to export instead of a few seconds because of the scaling operation.  Just being able to call scale_world_origin on all of the geometry should only take a few seconds.

I guess I could do what you are suggesting by writing the `scale` method it so that there's an `if origin is None` ... only multiple the vertices by the factor.  However, this means that there's an extra `if` with each vertex, which could start to add up when you have thousands of vertices in a mesh.  IF you think the extra minute or two that it could add to scaling a large mesh doesn't matter, then I'll change it to be as you suggest.

Reply:
> ... However, this means that there's an extra if with each vertex, which could start to add up when you have thousands of vertices in a mesh.

Not sure why you need to do it for each vertex. Can't you check it once and then have two different approaches?

if origin is None:
    # do this for all the vertices
else:
    # do something else.

This is a design decision mainly and I _think_ having one method is more elegant! Up to you.

Reply:
I agree one method is more elegant and maybe I am making too big of a deal out of the `if` calls. If I were to do what you are proposing, we would still need to keep the `scale_world_origin` method on the Point3D object or I would have to copy the code from this method into all of the other scale methods on the other objects.  Thinking about it a bit more, maybe it's ok to have this code copied everywhere as it's only one line.  Ok. I will implement it this way.